packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -72,129 +65,95 @@ const ListView = () => {
   const webhooksToDeleteLength = webhooksToDelete.length;
   const getWebhookIndex = (id) => webhooks.findIndex((webhook) => webhook.id === id);
 
-  useEffect(() => {
-    isMounted.current = true;
-
-    return () => {
-      isMounted.current = false;
-    };
-  }, []);
-
-  /**
-   * TODO: refactor this, but actually refactor
-   * the whole component. Needs some love.
-   */
-  useEffect(() => {
-    const fetchWebHooks = async () => {
-      try {
-        const {
-          data: { data },
-        } = await get('/admin/webhooks');
-
-        if (isMounted.current) {
-          dispatch({
-            type: 'GET_DATA_SUCCEEDED',
-            data,
-          });
-          notifyStatus('webhooks have been loaded');
-        }
-      } catch (err) {
-        console.log(err);
-
-        if (isMounted.current) {
-          if (err.code !== 20) {
-            toggleNotification({
-              type: 'warning',
-              message: { id: 'notification.error' },
-            });
-          }
-          dispatch({
-            type: 'TOGGLE_LOADING',
-          });
-        }
-      }
-    };
+  const fetchWebHooks = async () => {
+    try {
+      const {
+        data: { data },
+      } = await get('/admin/webhooks');
 
-    if (canRead) {
-      fetchWebHooks();
-    }
-  }, [canRead, get, notifyStatus, toggleNotification]);
+      setWebhooks(data);
+      notifyStatus('webhooks have been loaded');
 
-  const handleToggleModal = () => {
-    setShowModal((prev) => !prev);
-  };
+      return data;
+    } catch (err) {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error' },
```

Comment:
Can we please display the full error message here using the `useAPIErrorHandler` hook? https://contributor.strapi.io/hooks/use-api-error-handler